oneoffs/sharp_positions.py

```diff
@@ -0,0 +1,420 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Used as a starting point for our generating hard position collection.
+
+training_curve.py is a supervised approach, assuming the pro moves are correct
+Performance is then measured against the played move and eventual outcome.
+
+<NAME> is unsupervised, it tries to determine what the correct outcome and move
+is based on clustering of the strongest rated algorithms.
+"""
+
+import sys
+sys.path.insert(0, '.')
+
+import itertools
+import json
+import multiprocessing
+import os
+import random
+from collections import defaultdict, Counter
+
+import fire
+import numpy as np
+import tensorflow as tf
+from absl import flags
+from sklearn import svm
+from tqdm import tqdm
+
+import oneoff_utils
+
+# Filter tensorflow info
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '1'
+
+flags.DEFINE_integer("num_positions", 1, "How many positions from each game.")
+flags.DEFINE_integer("top_n", 3, "Policy moves to record per position.")
+flags.DEFINE_integer("min_idx", 100, "Min model number to include.")
+
+flags.DEFINE_string("sgf_dir", "data/s/", "collection of SGFs")
+flags.DEFINE_string("model_dir", "models/", "Folder of models")
+flags.DEFINE_string("rating_json", "ratings.json", "ratings of models")
+flags.DEFINE_string("results", "results.csv", "Evaluate results file")
+flags.DEFINE_string("SVM_json", "SVM_data.json", "SVM data about positions")
```

Comment:
your docstrings here should make clear the difference between inputs and outputs, even if its just from tenses.  "Evaluate results file" is ambiguous